Adding Springboot2 SSE

vijaycse opened the pull request:
This closes the issue https://github.com/resilience4j/resilience4j/issues/1076

vijaycse commented:
@RobWin  here is the PR to add SSE for springboot2. With this PR, both ratpack and springboot2 emits Identical stream events and hystrix stream events

vijaycse commented:
@RobWin  fixed CI issue. should be good for review
Thanks

dlsrb6342 commented on resilience4j-spring-boot2/build.gradle:
```diff
@@ -1,6 +1,9 @@
 dependencies {
     compile project(':resilience4j-spring')
     compile project(':resilience4j-micrometer')
+    compile project(':resilience4j-reactor')
+    compile(libraries.spring_boot2_webflux)
```
I think these should be `compileOnly`. 
There are users who doesn't use reactor or webflux.
What do you think?

vijaycse replied:
I see your point.. rxjava is already part of testCompile.  I could use Flowable instead of FLUX (thus webflux).  
rxjava is anyway part of the manifest of springboot2.  Would that be OK ?

dlsrb6342 replied:
Flowable also cannot be used because rxjava is only used in test. 
I think SSE endpoint should be only enabled when Flux exists by `ConditionalOnClass(Flux.class)`.

vijaycse replied:
makes sense to me. will let you know once the changes are done.

vijaycse replied:
Updated it with conditiononclass

dlsrb6342 replied:
Need to be `compileOnly`. Please change these :)

vijaycse replied:
done.  @dlsrb6342 . Thank you

dlsrb6342 commented on resilience4j-spring-boot2/build.gradle:
```diff
@@ -1,6 +1,9 @@
     compile project(':resilience4j-micrometer')
+    compile project(':resilience4j-reactor')
+    compile(libraries.spring_boot2_webflux)
+    compileOnly(libraries.rxjava2)
```
Is this needed?

vijaycse replied:
you are right. I forgot to remove this at the end. Will remove

vijaycse commented:
will take a look.

On Thu, Jul 23, 2020 at 9:46 PM Ingyu Hwang <notifications@github.com>
wrote:

> *@dlsrb6342* requested changes on this pull request.
> ------------------------------
>
> In resilience4j-spring-boot2/build.gradle
> <https://github.com/resilience4j/resilience4j/pull/1093#discussion_r459826587>
> :
>
> > +    compile project(':resilience4j-reactor')
> +    compile(libraries.spring_boot2_webflux)
>
> I think these should be compileOnly.
> There are users who doesn't use reactor or webflux.
> What do you think?
> ------------------------------
>
> In resilience4j-spring-boot2/build.gradle
> <https://github.com/resilience4j/resilience4j/pull/1093#discussion_r459826702>
> :
>
> > @@ -1,6 +1,9 @@
>  dependencies {
>      compile project(':resilience4j-spring')
>      compile project(':resilience4j-micrometer')
> +    compile project(':resilience4j-reactor')
> +    compile(libraries.spring_boot2_webflux)
> +    compileOnly(libraries.rxjava2)
>
> Is this needed?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/resilience4j/resilience4j/pull/1093#pullrequestreview-454607586>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDARNFY3EJ5JJ2DIE3OCDTR5DYYZANCNFSM4PFXIG6Q>
> .
>


-- 
Regards,
Vijay R

vijaycse commented:
@dlsrb6342  Can you review it again? Or do you think we need to add this as separate module ?

dlsrb6342 commented on resilience4j-spring-boot2/build.gradle:
```diff
@@ -1,7 +1,9 @@
 dependencies {
     compile project(':resilience4j-spring')
     compile project(':resilience4j-micrometer')
+    compile project(':resilience4j-reactor')
```
This also need to be `compileOnly`. 
The users who doesn't use reactor don't want to have this dependency.

vijaycse replied:
i see. made changes. thanks for correcting.

vijaycse replied:
@dlsrb6342

dlsrb6342 commented on resilience4j-spring-boot2/src/main/java/io/github/resilience4j/circuitbreaker/monitoring/endpoint/CircuitBreakerHystrixServerSideEvent.java:
```diff
@@ -0,0 +1,128 @@
+
+    private CircuitBreaker getCircuitBreaker(String circuitBreakerName) {
+        CircuitBreaker circuitBreaker = circuitBreakerRegistry.circuitBreaker(circuitBreakerName);
+        if (circuitBreaker == null && !circuitBreaker.getName().equalsIgnoreCase(circuitBreakerName)) {
```
I think this doesn't need to be checked. 
`CircuitBreakerRegistry` create or get when you request a circuitbreaker. 
Therefore, it cannot be null, and it must be the same as the requested name.

dlsrb6342 commented on resilience4j-spring-boot2/src/main/java/io/github/resilience4j/circuitbreaker/monitoring/endpoint/CircuitBreakerServerSideEvent.java:
```diff
@@ -0,0 +1,115 @@
+
+    private CircuitBreaker getCircuitBreaker(String circuitBreakerName) {
+        CircuitBreaker circuitBreaker = circuitBreakerRegistry.circuitBreaker(circuitBreakerName);
+        if (circuitBreaker == null && !circuitBreaker.getName().equalsIgnoreCase(circuitBreakerName)) {
```
I think this doesn't need to be checked. 
`CircuitBreakerRegistry` create or get when you request a circuitbreaker. 
Therefore, it cannot be null, and it must be the same as the requested name.

dlsrb6342 commented on resilience4j-spring-boot2/src/main/java/io/github/resilience4j/circuitbreaker/monitoring/endpoint/CircuitBreakerServerSideEvent.java:
```diff
@@ -0,0 +1,115 @@
+    }
+
+    private Function<CircuitBreakerEvent, String> getCircuitBreakerEventStringFunction() {
+        ObjectMapper jsonMapper = new ObjectMapper();
```
Can be moved to class member variable.

dlsrb6342 commented on resilience4j-spring-boot2/src/main/java/io/github/resilience4j/circuitbreaker/monitoring/endpoint/CircuitBreakerHystrixServerSideEvent.java:
```diff
@@ -0,0 +1,128 @@
+    }
+
+    private BiFunction<CircuitBreakerEvent, CircuitBreaker, String> getCircuitBreakerEventStringFunction() {
+        ObjectMapper jsonMapper = new ObjectMapper();
```
Can be moved to class member variable.

vijaycse commented:
good call outs. will work on it. CI is failing for some reason in spring
cloud that im not able to reproduce locally .
Im triaging that. Will get back to you

On Wed, Jul 29, 2020 at 8:59 PM Ingyu Hwang <notifications@github.com>
wrote:

> *@dlsrb6342* commented on this pull request.
> ------------------------------
>
> In
> resilience4j-spring-boot2/src/main/java/io/github/resilience4j/circuitbreaker/monitoring/endpoint/CircuitBreakerServerSideEvent.java
> <https://github.com/resilience4j/resilience4j/pull/1093#discussion_r462687829>
> :
>
> > +        CircuitBreaker circuitBreaker = getCircuitBreaker(name);
> +        Flux<CircuitBreakerEvent> eventStream = toFlux(circuitBreaker.getEventPublisher())
> +            .filter(
> +                event -> event.getEventType() == CircuitBreakerEvent.Type.valueOf(eventType.toUpperCase())
> +            );
> +        Function<CircuitBreakerEvent, String> data = getCircuitBreakerEventStringFunction();
> +        return eventStream.map(cbEvent -> ServerSentEvent.<String>builder()
> +            .id(cbEvent.getCircuitBreakerName())
> +            .event(cbEvent.getEventType().name())
> +            .data(data.apply(cbEvent))
> +            .build()
> +        );
> +    }
> +
> +    private Function<CircuitBreakerEvent, String> getCircuitBreakerEventStringFunction() {
> +        ObjectMapper jsonMapper = new ObjectMapper();
>
> Can be moved to class member variable.
> ------------------------------
>
> In
> resilience4j-spring-boot2/src/main/java/io/github/resilience4j/circuitbreaker/monitoring/endpoint/CircuitBreakerHystrixServerSideEvent.java
> <https://github.com/resilience4j/resilience4j/pull/1093#discussion_r462689496>
> :
>
> > +        CircuitBreaker circuitBreaker = getCircuitBreaker(name);
> +        Flux<CircuitBreakerEvent> eventStream = toFlux(circuitBreaker.getEventPublisher())
> +            .filter(
> +                event -> event.getEventType() == CircuitBreakerEvent.Type.valueOf(eventType.toUpperCase())
> +            );
> +        BiFunction<CircuitBreakerEvent, CircuitBreaker, String> data = getCircuitBreakerEventStringFunction();
> +        return eventStream.map(cbEvent -> ServerSentEvent.<String>builder()
> +            .id(cbEvent.getCircuitBreakerName())
> +            .event(cbEvent.getEventType().name())
> +            .data(data.apply(cbEvent, circuitBreaker))
> +            .build()
> +        );
> +    }
> +
> +    private BiFunction<CircuitBreakerEvent, CircuitBreaker, String> getCircuitBreakerEventStringFunction() {
> +        ObjectMapper jsonMapper = new ObjectMapper();
>
> Can be moved to class member variable.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/resilience4j/resilience4j/pull/1093#pullrequestreview-458018732>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDARNFVU3TZYCHRQCUB6LDR6DHYFANCNFSM4PFXIG6Q>
> .
>


-- 
Regards,
Vijay R

vijaycse commented:
@dlsrb6342  I have addressed the comments and fixed CI issue as well. Introduced property to enable and disable stream.event. I will update the docs once the PR is merged if i get access or will follow up with Rob to update the property in the doc later.
Thanks for the patience.

vijaycse commented on resilience4j-spring-boot2/src/main/java/io/github/resilience4j/circuitbreaker/monitoring/endpoint/CircuitBreakerHystrixServerSideEvent.java:
```diff
@@ -0,0 +1,139 @@
+
+    private CircuitBreaker getCircuitBreaker(String circuitBreakerName) {
+        CircuitBreaker circuitBreaker = circuitBreakerRegistry.circuitBreaker(circuitBreakerName);
+        if (!circuitBreaker.getName().equalsIgnoreCase(circuitBreakerName)) {
```
yeap. makes sense. pushed my changes

vijaycse commented on resilience4j-spring-boot2/src/main/java/io/github/resilience4j/circuitbreaker/autoconfigure/CircuitBreakerStreamEventsAutoConfiguration.java:
```diff
@@ -0,0 +1,54 @@
+@Configuration
+@ConditionalOnClass({CircuitBreaker.class, Endpoint.class})
+@AutoConfigureAfter(CircuitBreakerAutoConfiguration.class)
+@ConditionalOnProperty(value = "resilience4j.circuitBreaker.stream.event.enabled")
```
Agreed. Removed the property 
Tested locally with ```"management.endpoints.web.exposure.exclude: stream-circuitbreaker-events"```

vijaycse commented:
@dlsrb6342  Addressed your comments
Thanks

dlsrb6342 commented:
@RobWin Could you have a look this PR?

RobWin commented on resilience4j-spring-boot2/src/test/java/io/github/resilience4j/circuitbreaker/monitoring/events/CircuitBreakerHystrixStreamEventsTest.java:
```diff
@@ -0,0 +1,148 @@
+        dummyService.doSomething(false);
+
+        CircuitBreakerEventsEndpointResponse circuitBreakerEventsAfter = circuitBreakerEvents(ACTUATOR_CIRCUITBREAKEREVENTS);
+        assert (circuitBreakerEventsBefore.getCircuitBreakerEvents().size() < circuitBreakerEventsAfter.getCircuitBreakerEvents().size());
```
Please use AssertJ

RobWin commented on resilience4j-spring-boot2/src/test/java/io/github/resilience4j/circuitbreaker/monitoring/events/CircuitBreakerHystrixStreamEventsTest.java:
```diff
@@ -0,0 +1,148 @@
+        CircuitBreakerEventsEndpointResponse circuitBreakerEventsAfter = circuitBreakerEvents(ACTUATOR_CIRCUITBREAKEREVENTS);
+        assert (circuitBreakerEventsBefore.getCircuitBreakerEvents().size() < circuitBreakerEventsAfter.getCircuitBreakerEvents().size());
+        Thread.sleep(1000); // for webClient to complete the subscribe operation
+        assert (events.size() == 2);
```
Please use AssertJ

RobWin commented on resilience4j-spring-boot2/src/test/java/io/github/resilience4j/circuitbreaker/monitoring/events/CircuitBreakerHystrixStreamEventsTest.java:
```diff
@@ -0,0 +1,148 @@
+        }
+        CircuitBreakerEventsEndpointResponse circuitBreakerEventsAfter = circuitBreakerEvents(ACTUATOR_CIRCUITBREAKEREVENTS + "/backendA");
+        Thread.sleep(1000); //  webClient to complete the subscribe operation
+        assert (circuitBreakerEventsBefore.getCircuitBreakerEvents().size() < circuitBreakerEventsAfter.getCircuitBreakerEvents().size());
```
Please use AssertJ

RobWin commented on resilience4j-spring-boot2/src/test/java/io/github/resilience4j/circuitbreaker/monitoring/events/CircuitBreakerStreamEventsTest.java:
```diff
@@ -0,0 +1,149 @@
+        dummyService.doSomething(false);
+
+        CircuitBreakerEventsEndpointResponse circuitBreakerEventsAfter = circuitBreakerEvents(ACTUATOR_CIRCUITBREAKEREVENTS);
+        assert (circuitBreakerEventsBefore.getCircuitBreakerEvents().size() < circuitBreakerEventsAfter.getCircuitBreakerEvents().size());
```
Please use AssertJ

vijaycse commented:
@RobWin  I addressed your comments about assertJ.  Can you take a look? Thanks

RobWin commented on resilience4j-spring-boot2/src/test/java/io/github/resilience4j/circuitbreaker/monitoring/events/CircuitBreakerHystrixStreamEventsTest.java:
```diff
@@ -0,0 +1,148 @@
+        CircuitBreakerEventsEndpointResponse circuitBreakerEventsAfter = circuitBreakerEvents(ACTUATOR_CIRCUITBREAKEREVENTS + "/backendA");
+        Thread.sleep(1000); // for webClient to complete the subscribe operation
+        assertThat (circuitBreakerEventsBefore.getCircuitBreakerEvents().size() < circuitBreakerEventsAfter.getCircuitBreakerEvents().size());
+        assertThat (events.size() == 1);
```
Assertj has a fluent API. You have to do
`assertThat(events.size()).isEqulTo(1);`

vijaycse replied:
Oh.. yeah. I see.. I updated it and pushed it

vijaycse replied:
@RobWin

vijaycse commented:
@RobWin  Addressed your comments on using assertJ fluentAPI

RobWin commented:
Thx.
Are the `Thread.sleep` calls really needed?

vijaycse commented:
Good question. They are working locally fine but CI is failing without
Thread.sleep. It appears there is some pause or wait time for the changes
to reflect espcially during subscribing the event..

On Fri, Aug 14, 2020 at 7:08 AM Robert Winkler <notifications@github.com>
wrote:

> Thx.
> Are the Thread.sleep calls really needed?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/resilience4j/resilience4j/pull/1093#issuecomment-674043127>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDARNGJ5LWIUJZCXN7DFZ3SAUSKZANCNFSM4PFXIG6Q>
> .
>


-- 
Regards,
Vijay R

vijaycse commented:
@RobWin  I addressed your comments. Please let me know if any questions.

RobWin commented on resilience4j-spring-boot2/src/test/java/io/github/resilience4j/circuitbreaker/monitoring/events/CircuitBreakerHystrixStreamEventsTest.java:
```diff
@@ -0,0 +1,148 @@
+    }
+
+    private Flux<ServerSentEvent<String>> circuitBreakerStreamEvents(String s) {
+        WebClient client = WebClient.create("http://localhost:" + randomServerPort);
```
Is it possible to use the WebTestClient  here?

Here is a guide how to use the WebTestClient for streaming responses: 
https://docs.spring.io/spring/docs/current/spring-framework-reference/pdf/testing-webtestclient.pdf

Then please use the `StepVerifier` to subscribe to the stream and verify that the streams emits one event.

RobWin replied:
Yes, I think the problem is that `.exchange()` is blocking :( 
Internally it's doing
``` 
ClientResponse clientResponse = this.bodySpec.exchange().block(getTimeout());
```

RobWin commented on resilience4j-spring-boot2/src/test/java/io/github/resilience4j/circuitbreaker/monitoring/events/CircuitBreakerStreamEventsTest.java:
```diff
@@ -0,0 +1,150 @@
+    }
+
+    private Flux<ServerSentEvent<String>> circuitBreakerStreamEvents(String s) {
+        WebClient client = WebClient.create("http://localhost:" + randomServerPort);
```
Is it possible to use the WebTestClient here?
Here is a guide how to use the WebTestClient for streaming responses: 
https://docs.spring.io/spring/docs/current/spring-framework-reference/pdf/testing-webtestclient.pdf

Then please use the `StepVerifier` to subscribe to the stream and verify that the streams emits one event.

vijaycse commented:
I will look at it
Thanks

On Wed, Aug 19, 2020 at 1:16 AM Robert Winkler <notifications@github.com>
wrote:

> *@RobWin* requested changes on this pull request.
> ------------------------------
>
> In
> resilience4j-spring-boot2/src/test/java/io/github/resilience4j/circuitbreaker/monitoring/events/CircuitBreakerHystrixStreamEventsTest.java
> <https://github.com/resilience4j/resilience4j/pull/1093#discussion_r472723457>
> :
>
> > +    @LocalServerPort
> +    int randomServerPort;
> +
> +    @Autowired
> +    DummyService dummyService;
> +
> +    @Test
> +    public void streamAllEvents() throws IOException, InterruptedException {
> +        List<ServerSentEvent<String>> events = getServerSentEvents(ACTUATOR_STREAM_CIRCUITBREAKER_EVENTS);
> +        CircuitBreakerEventsEndpointResponse circuitBreakerEventsBefore = circuitBreakerEvents(ACTUATOR_CIRCUITBREAKEREVENTS);
> +        try {
> +            dummyService.doSomething(true);
> +        } catch (IOException ex) {
> +            // Do nothing. The IOException is recorded by the CircuitBreaker as part of the recordFailurePredicate as a failure.
> +        }
> +        Thread.sleep(1000); // to record the event
>
> Waiting between calls should not be necessary
> ------------------------------
>
> In
> resilience4j-spring-boot2/src/test/java/io/github/resilience4j/circuitbreaker/monitoring/events/CircuitBreakerHystrixStreamEventsTest.java
> <https://github.com/resilience4j/resilience4j/pull/1093#discussion_r472723524>
> :
>
> > +        // The invocation is recorded by the CircuitBreaker as a success.
> +        dummyService.doSomething(false);
> +
> +        CircuitBreakerEventsEndpointResponse circuitBreakerEventsAfter = circuitBreakerEvents(ACTUATOR_CIRCUITBREAKEREVENTS);
> +        assertThat (circuitBreakerEventsBefore.getCircuitBreakerEvents().size()).isLessThan(circuitBreakerEventsAfter.getCircuitBreakerEvents().size());
> +        Thread.sleep(1000); // for webClient to complete the subscribe operation
> +        assertThat (events.size()).isEqualTo(2);
> +    }
> +
> +    @Test
> +    public void streamEventsbyName() throws IOException, InterruptedException {
> +        List<ServerSentEvent<String>> events = getServerSentEvents(ACTUATOR_STREAM_CIRCUITBREAKER_EVENTS + "/backendA");
> +        CircuitBreakerEventsEndpointResponse circuitBreakerEventsBefore = circuitBreakerEvents(ACTUATOR_CIRCUITBREAKEREVENTS + "/backendA");
> +        // The invocation is recorded by the CircuitBreaker as a success.
> +        dummyService.doSomething(false);
> +        Thread.sleep(1000); // sleep is needed to record the event
>
> Waiting between calls should not be necessary
> ------------------------------
>
> In
> resilience4j-spring-boot2/src/test/java/io/github/resilience4j/circuitbreaker/monitoring/events/CircuitBreakerHystrixStreamEventsTest.java
> <https://github.com/resilience4j/resilience4j/pull/1093#discussion_r472723558>
> :
>
> > +        } catch (IOException ex) {
> +            // Do nothing. The IOException is recorded by the CircuitBreaker as part of the recordFailurePredicate as a failure.
> +        }
> +        CircuitBreakerEventsEndpointResponse circuitBreakerEventsAfter = circuitBreakerEvents(ACTUATOR_CIRCUITBREAKEREVENTS + "/backendA");
> +        Thread.sleep(1000); //  webClient to complete the subscribe operation
> +        assertThat (circuitBreakerEventsBefore.getCircuitBreakerEvents().size()).isLessThan(circuitBreakerEventsAfter.getCircuitBreakerEvents().size());
> +        assertThat (events.size()).isEqualTo(2);
> +    }
> +
> +    @Test
> +    public void streamEventsbyNameAndType() throws IOException, InterruptedException {
> +        List<ServerSentEvent<String>> events = getServerSentEvents(ACTUATOR_STREAM_CIRCUITBREAKER_EVENTS+ "/backendA/ERROR");
> +        CircuitBreakerEventsEndpointResponse circuitBreakerEventsBefore = circuitBreakerEvents(ACTUATOR_CIRCUITBREAKEREVENTS + "/backendA");
> +        // The invocation is recorded by the CircuitBreaker as a success.
> +        dummyService.doSomething(false);
> +        Thread.sleep(1000); // to record the event
>
> Waiting between calls should not be necessary
> ------------------------------
>
> In
> resilience4j-spring-boot2/src/test/java/io/github/resilience4j/circuitbreaker/monitoring/events/CircuitBreakerHystrixStreamEventsTest.java
> <https://github.com/resilience4j/resilience4j/pull/1093#discussion_r472743584>
> :
>
> > +            content -> events.add(content),
> +            error -> System.out.println("Error receiving SSE: {}" + error),
> +            () -> System.out.println("Completed!!!"));
> +        return events;
> +    }
> +
> +    private CircuitBreakerEventsEndpointResponse circuitBreakerEvents(String s) {
> +        return this.webTestClient.get().uri(s).exchange()
> +            .expectStatus().isOk()
> +            .expectBody(CircuitBreakerEventsEndpointResponse.class)
> +            .returnResult()
> +            .getResponseBody();
> +    }
> +
> +    private Flux<ServerSentEvent<String>> circuitBreakerStreamEvents(String s) {
> +        WebClient client = WebClient.create("http://localhost:" + randomServerPort);
>
> Is it possible to use the WebTestClient here?
>
> Here is a guide how to use the WebTestClient for streaming responses:
>
> https://docs.spring.io/spring/docs/current/spring-framework-reference/pdf/testing-webtestclient.pdf
>
> Then please use the StepVerifier to subscribe to the stream and verify
> that the streams emits one event.
> ------------------------------
>
> In
> resilience4j-spring-boot2/src/test/java/io/github/resilience4j/circuitbreaker/monitoring/events/CircuitBreakerStreamEventsTest.java
> <https://github.com/resilience4j/resilience4j/pull/1093#discussion_r472743662>
> :
>
> > +            content -> events.add(content),
> +            error -> System.out.println("Error receiving SSE: {}" + error),
> +            () -> System.out.println("Completed!!!"));
> +        return events;
> +    }
> +
> +    private CircuitBreakerEventsEndpointResponse circuitBreakerEvents(String s) {
> +        return this.webTestClient.get().uri(s).exchange()
> +            .expectStatus().isOk()
> +            .expectBody(CircuitBreakerEventsEndpointResponse.class)
> +            .returnResult()
> +            .getResponseBody();
> +    }
> +
> +    private Flux<ServerSentEvent<String>> circuitBreakerStreamEvents(String s) {
> +        WebClient client = WebClient.create("http://localhost:" + randomServerPort);
>
> Is it possible to use the WebTestClient here?
> Here is a guide how to use the WebTestClient for streaming responses:
>
> https://docs.spring.io/spring/docs/current/spring-framework-reference/pdf/testing-webtestclient.pdf
>
> Then please use the StepVerifier to subscribe to the stream and verify
> that the streams emits one event.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/resilience4j/resilience4j/pull/1093#pullrequestreview-470139444>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDARNA4MXDFIDZDNOKQERTSBNU4BANCNFSM4PFXIG6Q>
> .
>


-- 
Regards,
Vijay R

RobWin commented:
Thanks. Could you please add some documentation below the "Events Endpoint" chapter here -> https://resilience4j.readme.io/docs/getting-started-3?
You can press the button "Suggest Edits".

RobWin commented:
Damn the test `CircuitBreakerHystrixStreamEventsTest` fails on master

vijaycse commented:
let me take a look

On Thu, Aug 27, 2020 at 9:26 AM Robert Winkler <notifications@github.com>
wrote:

> Damn the test CircuitBreakerHystrixStreamEventsTest fails on master.
>
> io.github.resilience4j.circuitbreaker.monitoring.events.CircuitBreakerHystrixStreamEventsTest > streamEventsbyName FAILED
>     org.opentest4j.AssertionFailedError at CircuitBreakerHystrixStreamEventsTest.java:92
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/resilience4j/resilience4j/pull/1093#issuecomment-681983534>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDARNG5662OWIJF3N7UEILSCZUKNANCNFSM4PFXIG6Q>
> .
>


-- 
Regards,
Vijay R

vijaycse commented:
I made a temp fix while working on the permanent fix.  Please see

https://github.com/resilience4j/resilience4j/pull/1146/files



On Thu, Aug 27, 2020 at 9:27 AM Vijayalayan Ramalingam <
vijay.ramcse@gmail.com> wrote:

> let me take a look
>
> On Thu, Aug 27, 2020 at 9:26 AM Robert Winkler <notifications@github.com>
> wrote:
>
>> Damn the test CircuitBreakerHystrixStreamEventsTest fails on master.
>>
>> io.github.resilience4j.circuitbreaker.monitoring.events.CircuitBreakerHystrixStreamEventsTest > streamEventsbyName FAILED
>>     org.opentest4j.AssertionFailedError at CircuitBreakerHystrixStreamEventsTest.java:92
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/resilience4j/resilience4j/pull/1093#issuecomment-681983534>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ABDARNG5662OWIJF3N7UEILSCZUKNANCNFSM4PFXIG6Q>
>> .
>>
>
>
> --
> Regards,
> Vijay R
>


-- 
Regards,
Vijay R

vijaycse commented:
This is also failing.  Running fine locally. looking at it..

On Thu, Aug 27, 2020 at 8:25 PM Vijayalayan Ramalingam <
vijay.ramcse@gmail.com> wrote:

> I made a temp fix while working on the permanent fix.  Please see
>
> https://github.com/resilience4j/resilience4j/pull/1146/files
>
>
>
> On Thu, Aug 27, 2020 at 9:27 AM Vijayalayan Ramalingam <
> vijay.ramcse@gmail.com> wrote:
>
>> let me take a look
>>
>> On Thu, Aug 27, 2020 at 9:26 AM Robert Winkler <notifications@github.com>
>> wrote:
>>
>>> Damn the test CircuitBreakerHystrixStreamEventsTest fails on master.
>>>
>>> io.github.resilience4j.circuitbreaker.monitoring.events.CircuitBreakerHystrixStreamEventsTest > streamEventsbyName FAILED
>>>     org.opentest4j.AssertionFailedError at CircuitBreakerHystrixStreamEventsTest.java:92
>>>
>>> —
>>> You are receiving this because you authored the thread.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/resilience4j/resilience4j/pull/1093#issuecomment-681983534>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/ABDARNG5662OWIJF3N7UEILSCZUKNANCNFSM4PFXIG6Q>
>>> .
>>>
>>
>>
>> --
>> Regards,
>> Vijay R
>>
>
>
> --
> Regards,
> Vijay R
>


-- 
Regards,
Vijay R

RobWin commented:
Maybe change the test to test against the Controller and use the StepVerifier as suggested in your comment?
